crates/jsonschema/src/keywords/helpers.rs

```diff
@@ -27,9 +31,11 @@ pub(crate) fn map_get_u64<'a>(
                     return Some(Ok(value as u64));
                 }
             }
+            let location = ctx.location().join(type_name);
             Some(Err(ValidationError::single_type_error(
```

Comment:
This type of validation errors is a safety net - usually it should be handled with metaschema validation. I.e. they replicate metaschema behavior and as it is handled by some keyword in the metaschema, these errors should have (I think in most cases) proper location.

Reply:
Or maybe just remove those completely if they are truly unreachable